lib/src/widget/nav_bar/widget/nav_bar_container.dart

```diff
@@ -0,0 +1,144 @@
+final _containerNotImplementedException =
+    Exception('Did you forget to add $NavBarContainer?\n'
+        'Please wrap your MaterialApp with $NavBarContainer');
+
+class NavBarContainer extends StatefulWidget {
+  final Widget child;
+
+  const NavBarContainer({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  _NavBarContainerState createState() => _NavBarContainerState();
+
+  /// Call this, when you need to update current instance of the [NavBar]
+  /// f.e. [NavBarConfig] was changed for the screen
+  static void updateNavBar(BuildContext context) {
+    final typedWidget =
+        context.dependOnInheritedWidgetOfExactType<_InheritedWidget>();
+    if (typedWidget == null) {
+      throw _containerNotImplementedException;
+    }
+    return typedWidget.controller.updateNavBar();
+  }
+
+  /// This method used internally (from [NavBarObserver])
+  /// Will initiate walk through the children widgets,
+  /// find the latest instance of [NavBarConfig] and if it is not [null],
+  /// then will draw [NavBar] with it.
+  /// If not [NavBarConfig] found - [NavBar] will be not shown(removed)
+  ///
+  /// [goingBack] is a flag, that indicates that navigation flow going back.
+  /// when [true], then will ignore latest [NavBarConfig] (from screen that will be removed)
+  /// and will try to apply previous [NavBarConfig] (from screen that will be shown instead)
+  static void resetNavBar(
+    BuildContext context, {
+    required bool goingBack,
+  }) {
+    final typedWidget =
+        context.dependOnInheritedWidgetOfExactType<_InheritedWidget>();
+    if (typedWidget == null) {
+      throw _containerNotImplementedException;
+    }
+    return typedWidget.controller.resetNavBar(goingBack);
+  }
+}
+
+class _NavBarContainerState extends State<NavBarContainer>
+    implements _NavBarController {
+  late ConfigPair _configPair;
+
+  Linden get linden => UnterDenLinden.getLinden(context);
+
+  @override
+  Widget build(BuildContext context) =>
+      _InheritedWidget(controller: this, child: widget.child);
+
+  @override
+  void resetNavBar(bool goingBack) {
+    WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
+      if (!mounted) return;
+      _configPair = _getNavBarConfigPairToShow(context, goingBack);
+      _updateBar();
+    });
+  }
+
+  @override
+  void updateNavBar() {
+    WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
+      if (!mounted) return;
+      _updateBar();
+    });
+  }
+
+  void _updateBar() {
+    final config = _configPair.mixin?.navBarConfig;
+    _configPair.updater.update(config);
+  }
+
+  ConfigPair _getNavBarConfigPairToShow(
+    BuildContext context,
+    bool ignoreLast,
+  ) {
+    NavBarConfigMixin? preLastMixin;
+    NavBarConfigMixin? lastMixin;
+    ConfigUpdater? updater;
+    void visitor(Element element) {
+      if (element.widget is NavBar) {
+        final widget = element as StatefulElement;
+        updater = widget.state as ConfigUpdater;
```

Comment:
💅  to be really strict I would throw another exception if we find a second navBar

Reply:
yep, make sense 😉